hibernate-core/src/main/java/org/hibernate/type/AbstractType.java

```diff
@@ -134,9 +135,14 @@ public Object replace(
 			ForeignKeyDirection foreignKeyDirection) 
 	throws HibernateException {
 		boolean include;
-		if ( isAssociationType() ) {
+		if ( isAssociationType()) {
 			AssociationType atype = (AssociationType) this;
-			include = atype.getForeignKeyDirection()==foreignKeyDirection;
+			if ( atype instanceof OneToOneType && atype.useLHSPrimaryKey()) {
```

Comment:
I think the source of the bug is that the foreign key direction for ```atype``` is incorrect. It should be ```FROM_PARENT```, but it is ```TO_PARENT```.

I'm seeing both sides of the bidirectional associations have foreign key direction set to ```TO_PARENT```, which is clearly incorrect.

If the foreign key direction for ```Segunda#parent``` were correct, this fix would not be necessary.

I think the proper fix would be to fix the foreign key direction instead of disregarding it.

Reply:
@gbadner Thanks, Gail. I'll investigate to see why the `getForeignKeyDirection()` returns `TO_PARENT` then.